lib/generic/modules/metabolic_syndrome_care.json

```diff
@@ -0,0 +1,1857 @@
+                  "code" : "157141000119108",
+                  "display" : "Proteinuria due to type 2 diabetes mellitus (disorder)"
+                }]
+              }
+            ]
+          },
+          "transition" : "End_Proteinuria"
+        },
+        {
+          "transition" : "Check_End_Stage_Renal_Disease"
+        }
+      ]
+    },
+
+    "Diagnose_Proteinuria" : {
+      "type" : "ConditionOnset",
+      "codes" : [{
+        "system" : "SNOMED-CT",
+        "code" : "157141000119108",
+        "display" : "Proteinuria due to type 2 diabetes mellitus (disorder)"
+      }],
+      "direct_transition" : "Check_End_Stage_Renal_Disease"
+    },
+
+    "End_Proteinuria" : {
+      "type" : "ConditionEnd",
+      "codes" : [{
+        "system" : "SNOMED-CT",
+        "code" : "157141000119108",
+        "display" : "Proteinuria due to type 2 diabetes mellitus (disorder)"
+      }],
+      "direct_transition" : "Check_End_Stage_Renal_Disease"
+    },
+
+
+    "Check_End_Stage_Renal_Disease" : {
+      "type" : "Simple",
+      "conditional_transition" : [
+        {
+          "condition" : {
+            "condition_type" : "And",
+            "conditions" : [
+              {
+                "condition_type" : "Attribute",
+                "attribute" : "end_stage_renal_disease",
+                "operator" : "is not nil"
+              },
+              {
+                "condition_type" : "Not",
+                "condition" : {
+                  "condition_type" : "Active Condition",
+                  "codes" : [{
+                    "system" : "SNOMED-CT",
+                    "code" : "46177005",
+                    "display" : "End stage renal disease (disorder)"
+                  }]
+                }
+              }
+            ]
+          },
+          "transition" : "Diagnose_End_Stage_Renal_Disease"
+        },
+        {
+          "condition" : {
+            "condition_type" : "And",
+            "conditions" : [
+              {
+                "condition_type" : "Attribute",
+                "attribute" : "end_stage_renal_disease",
+                "operator" : "is nil"
+              },
+              {
+                "condition_type" : "Active Condition",
+                "codes" : [{
+                  "system" : "SNOMED-CT",
+                  "code" : "46177005",
+                  "display" : "End stage renal disease (disorder)"
+                }]
+              }
+            ]
+          },
+          "transition" : "End_End_Stage_Renal_Disease"
+        },
+        {
+          "transition" : "Check_Retinopathy"
+        }
+      ]
+    },
+
+    "Diagnose_End_Stage_Renal_Disease" : {
+      "type" : "ConditionOnset",
+      "codes" : [{
+        "system" : "SNOMED-CT",
+        "code" : "46177005",
+        "display" : "End stage renal disease (disorder)"
+      }],
+      "direct_transition" : "Check_Retinopathy"
+    },
+
+    "End_End_Stage_Renal_Disease" : {
+      "type" : "ConditionEnd",
```

Comment:
I'm pretty sure the only way this ends is that you die or get a kidney transplant.